apps/accounts/views.py

```diff
@@ -71,7 +71,22 @@ def inner(self, request):
                                            config=settings.FXA_CONFIG)
         except verify.IdentificationError:
             return Response({'error': 'Profile not found.'}, status=401)
-        return fn(self, request, user=find_user(identity), identity=identity)
+        else:
+            identity_user = find_user(identity)
+            if request.user.is_authenticated():
```

Comment:
I definitely started implementing this stuff in `login_user()` at first and then realised that `login_user()` doesn't really care as long as it gets the right user.